fs/fs.go

```diff
@@ -190,6 +257,11 @@ func NewFilesystem(ctx context.Context, root string, cfg config.FSConfig, opts .
 		metrics.Register(ns) // Register layer metrics.
 	}
 
+	pr := newPreresolver(fsOpts.maxConcurrency)
+	if err := pr.Start(ctx); err != nil {
```

Comment:
nit: Since the preresolver doesn't return an error, I think we can just run pr.Start in a goroutine here:

```go
pr := newPreresolver(fsOpts.maxConcurrency)
go pr.Start(ctx)
```

Reply:
We can, I just wrapped it in that format in case we decide that Start ever needs to return an error. The only context I can think of now that would make sense, though, is if we pass in a bad context and want to be louder on the preresolver failing. But I think it might be better to add logging to show this, instead.